packages/mui-system/src/cssVars/createCssVarsProvider.js

```diff
@@ -64,7 +64,10 @@ export default function createCssVarsProvider(options) {
   }) {
     // make sure that baseTheme is always independent of each <CssVarsProvider /> call.
     // JSON.parse(JSON.stringify(...)) is okay to be used as long as the baseTheme is a plain object.
-    const clonedBaseTheme = React.useMemo(() => JSON.parse(JSON.stringify(baseTheme)), []);
+    // const clonedBaseTheme = React.useMemo(() => JSON.parse(JSON.stringify(baseTheme)), []);
+    // TODO: this is not working for MD theme where there are functions, like
+    // transitions.create, palette.getContrastText etc.
+    const clonedBaseTheme = React.useMemo(() => ({ ...baseTheme }), []);
```

Comment:
It took me some time till I found this :D

Reply:
😅 It's time to remove that line. Here is the fix https://github.com/mui/material-ui/pull/31148. In summary, the theme is not mutated anymore, so `clonedBaseTheme` can be removed.